src/msgpack.light/TokenBuilders/IMsgPackTokenBuilder.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace ProGaudi.MsgPack.Light.TokenBuilders
+{
+    internal interface IMsgPackTokenBuilder
+    {
+        void Add(DataTypes dataType);
+
+        MsgPackToken CreateToken(DataTypes dataType);
```

Comment:
Do we really need this construct? If someone has a type token and byte array, he will always able to deserialize it.